src/DGtal/dec/ATSolver2D.h

```diff
@@ -0,0 +1,1028 @@
+    std::tuple<double,double,double> checkV0() const
+    {
+      const double m1 = v0.myContainer.minCoeff();
+      const double m2 = v0.myContainer.maxCoeff();
+      const double ma = v0.myContainer.mean();
+      if ( verbose >= 1 )
+        trace.info() << "0-form v (should be in [0,1]): min=" << m1 << " avg=" << ma << " max=" << m2 << std::endl;
+      return std::make_tuple( m1, m2, ma );
+    }
+
+    
+    /// @return the discontinuity function v as a primal 0-form.
+    PrimalForm0 getV0() const
+    {
+      return v0;
+    }
+    
+    /// @return the discontinuity function v as a primal 1-form.
+    PrimalForm1 getV1() const
+    {
+      return M01*v0;
+    }
+
+    /// @return the discontinuity function u as a primal 2-form.
+    PrimalForm2 getV2() const
+    {
+      return M12*M01*v0;
+    }
+
+    /// @param k an integer such that `0 <= k < u2.size()`
+    /// @return the k-th piecewise smooth function u as a primal 2-form.
+    PrimalForm2 getU2( Dimension k ) const
+    {
+      return u2[ k ];
+    }
+    
+    /// Given a range of surfels [itB,itE), returns in \a output the
+    /// regularized vector field u.
+    ///
+    /// @tparam VectorFieldOutput the type of vector field for output values (RandomAccess container)
+    /// @tparam SurfelRangeConstIterator the type of iterator for traversing a range of surfels
+    ///
+    /// @param[out] output the vector of output values (a scalar or
+    /// vector field depending on input).
+    ///
+    /// @param itB the start of the range of surfels.
+    /// @param itE past the end of the range of surfels.
+    template <typename VectorFieldOutput, typename SurfelRangeConstIterator>
+    void
+    getOutputVectorFieldU2( VectorFieldOutput& output,
+                            SurfelRangeConstIterator itB, SurfelRangeConstIterator itE )
+    {
+      const Dimension N = u2.size();
+      Index i = 0;
+      for ( auto it = itB; it != itE; ++it, ++i )
+	{
+	  Index idx = surfel2idx[ *it ];
+          ASSERT( output[ i ].size() >= N );
+	  for ( Dimension k = 0; k < N; ++k )
+	    output[ i ][ k ] = u2[ k ].myContainer( idx );
+	}
+    }
+
+    /// Given a range of surfels [itB,itE), returns in \a output the
+    /// regularized scalar field u.
+    ///
+    /// @tparam ScalarFieldOutput the type of scalar field for output values (RandomAccess container)
+    /// @tparam SurfelRangeConstIterator the type of iterator for traversing a range of surfels
+    ///
+    /// @param[out] output the vector of output values (a scalar field), which should be of size `length(itB,itE)`
+    ///
+    /// @param itB the start of the range of surfels.
+    /// @param itE past the end of the range of surfels.
+    template <typename ScalarFieldOutput, typename SurfelRangeConstIterator>
+    void
+    getOutputScalarFieldU2( ScalarFieldOutput& output,
+                            SurfelRangeConstIterator itB, SurfelRangeConstIterator itE )
+    {
+      const Dimension N = u2.size();
```

Comment:
this will raise "unused variable" warnings in non Debug mode..

Reply:
You are right. Fixed